sycl/source/detail/event_impl.cpp

```diff
@@ -278,6 +278,13 @@ void event_impl::checkProfilingPreconditions() const {
         "Profiling information is unavailable as the queue associated with "
         "the event does not have the 'enable_profiling' property.");
   }
+  if (isEventFromSubmittedExecGraph() && (MExecGraph.lock()->getPartitions().size() > 1)) {
```

Comment:
need to check `MExecGraph.expired()` before dereferencing `MExecGraph.lock()`?
Ideally we'd use a pattern like this for locking the weak pointer (here and in other places)
```cpp
if (auto ExecGraphSP = MExecGraph.lock())  {
  ExecGraphSP->DoSomething();
}
```